core/src/com/unciv/logic/map/tile/TileImprovementFunctions.kt

```diff
@@ -311,31 +311,32 @@ class TileImprovementFunctions(val tile: Tile) {
         val closestCity = civ.cities.minByOrNull { it.getCenterTile().aerialDistanceTo(tile) }
             ?: return
         val distance = closestCity.getCenterTile().aerialDistanceTo(tile)
-        var productionPoints = 0f
+        var stats = Stats()
         for (unique in tile.getTerrainMatchingUniques(UniqueType.ProductionBonusWhenRemoved)) {
-            productionPoints = unique.params[0].toFloatOrNull() ?: return
+            //productionPoints = unique.params[0].toFloatOrNull() ?: return
+            stats.add(unique.stats)
         }
         val ruleset = civ.gameInfo.ruleset
         val choppingYieldsIncreaseWithGameProgress = ruleset.modOptions.constants.choppingYieldsIncreaseWithGameProgress
         // Civ6 yields increase with game progression: https://www.reddit.com/r/civ/comments/gvx44v/comment/fsrifc2/
         if (choppingYieldsIncreaseWithGameProgress) {
             val gameProgress = max(civ.tech.researchedTechnologies.size.toFloat() / ruleset.technologies.size, civ.policies.adoptedPolicies.size.toFloat() / ruleset.policies.size)
-            productionPoints *= (1 + 9 * gameProgress)
+            stats *= (1 + 9 * gameProgress)
+        }
+        if (distance != 1) stats *= (6-distance)/4
+        if (tile.owningCity == null || tile.owningCity!!.civ != civ) stats *= 2/3
+        stats *= civ.gameInfo.speed.productionCostModifier
+        for ((stat, value) in stats) {
+            if (closestCity != null) {
+                closestCity.addStat(stat, value.toInt())
+            }
         }
-        var productionPointsToAdd = if (distance == 1) productionPoints else productionPoints - (distance - 2) * productionPoints/4
-        if (tile.owningCity == null || tile.owningCity!!.civ != civ) productionPointsToAdd =
-            productionPointsToAdd * 2 / 3
-        productionPointsToAdd *= civ.gameInfo.speed.productionCostModifier
-        productionPointsToAdd = floor(productionPointsToAdd)
-        if (productionPointsToAdd > 0) {
-            closestCity.cityConstructions.addProductionPoints(productionPointsToAdd.toInt())
             val locations = LocationAction(tile.position, closestCity.location)
             civ.addNotification(
-                "Clearing a [$removedTerrainFeature] has created [${productionPointsToAdd.toInt()}] Production for [${closestCity.name}]",
+                "Clearing a [$removedTerrainFeature] has created [${stats.toStringWithoutIcons()}] for [${closestCity.name}]",
```

Comment:
This should be .toStringForNotifications, so it's translated well

Reply:
Are you sure about that? It's now showing black dots in front of the stat type.

Reply:
![Screenshot 2025-03-17 125638](https://github.com/user-attachments/assets/5a0083e9-41a0-45a5-94d1-3d4be4615dd6)

Reply:
Yes, otherwise it borks on language changes
Re: the image: That's interesting, can you send me the game save?
Is that in the notifications history?